solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder/src/main/java/com/solace/spring/cloud/stream/binder/SolaceMessageChannelBinder.java

```diff
@@ -120,6 +120,10 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination destination
 				getConsumerEndpointProperties(properties),
 				solaceMeterAccessor);
 
+		if (bindingsHealthContributor != null) {
+			adapter.setSolaceBindingsHealthContributor(bindingsHealthContributor);
+		}
+
```

Comment:
@Nephery please feel free to add that functionality there by yourself as you are the initial author of it. It will take way longer time going back and forth with comments here.